### PR TITLE
feat(ui-shell): blank right panel and switcher

### LINKED ISSUE #2403: Development of standalone Right Panel
Design issue: https://github.com/carbon-design-system/carbon-components/issues/2336

### PR DESCRIPTION
**Pull request by aledavila**
Closes #2403

This creates the react standalone blank right panel. 
Also react switcher component 

#### Changelog

**New**

- blank right panel
- switcher navigation built into right panel
- switcher styles file 
- color tokens for right panel

**Changed**

- created a new switcher in react not built on top of existing vanilla 

**Need to be Removed**

- need to remove vanilla 'product-switcher'

### CODE REVIEW

File: packages/react/src/components/UIShell/HeaderGlobalAction.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ const HeaderGlobalAction = ({
       {...rest}
       {...accessibilityLabel}
       className={className}
-      onClick={onClick}
+      onClick={event => console.log(event)}
```
Comment (asudoh) [inline packages/react/src/components/UIShell/HeaderGlobalAction.js]: Did you intend to replace this with something else?

File: packages/react/src/components/UIShell/HeaderPanel.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  };
+
+  const className = cx({
+    [`${prefix}--header-panel`]: true,
+    [`${prefix}--header-panel--expanded`]: expanded,
+    [customClassName]: !!customClassName,
+  });
+
+  return (
+    <div ref={ref} className={`${className}`} {...accessibilityLabel}>
```
Comment (asudoh) [inline packages/react/src/components/UIShell/HeaderPanel.js]:
```suggestion
    <div ref={ref} className={className} {...accessibilityLabel}>
```

File: packages/react/src/components/UIShell/HeaderPanel.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useState, useRef } from 'react';
```
Comment (asudoh) [inline packages/react/src/components/UIShell/HeaderPanel.js]:
```suggestion
import React from 'react';
```

Couldn't find `useState`/`useRef` references.

File: packages/react/src/components/UIShell/Switcher.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useState, useRef } from 'react';
```
Comment (asudoh) [inline packages/react/src/components/UIShell/Switcher.js]:
```suggestion
import React from 'react';
```

Couldn't find `useState`/`useRef` references.

File: packages/react/src/components/UIShell/Switcher.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    [`${prefix}--switcher--item-link`]: true,
+    [customClassName]: !!customClassName,
+  });
+
+  return (
+    <ul className={`${className}`} {...accessibilityLabel}>
+      {links &&
+        links.map((link, i) => {
+          return (
+            <li key={i} className={`${classNameItem}`}>
```
Comment (asudoh) [inline packages/react/src/components/UIShell/Switcher.js]:
```suggestion
            <li key={i} className={classNameItem}>
```

File: packages/react/src/components/UIShell/HeaderPanel.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    expanded,
+  } = props;
+
+  const accessibilityLabel = {
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledBy,
+  };
+
+  const className = cx({
+    [`${prefix}--header-panel`]: true,
```
Comment (asudoh) [inline packages/react/src/components/UIShell/HeaderPanel.js]:
```suggestion
```

File: packages/react/src/components/UIShell/HeaderPanel.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    className: customClassName,
+    expanded,
+  } = props;
+
+  const accessibilityLabel = {
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledBy,
+  };
+
+  const className = cx({
```
Comment (asudoh) [inline packages/react/src/components/UIShell/HeaderPanel.js]:
```suggestion
  const className = cx(`${prefix}--header-panel`, {
```

File: packages/react/src/components/UIShell/Switcher.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    'aria-labelledby': ariaLabelledBy,
+  };
+
+  const className = cx({
+    [`${prefix}--switcher`]: true,
+    [customClassName]: !!customClassName,
+  });
+
+  const classNameItem = cx({
+    [`${prefix}--switcher--item`]: true,
```
Comment (asudoh) [inline packages/react/src/components/UIShell/Switcher.js]:
```suggestion
```

File: packages/react/src/components/UIShell/Switcher.js [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledBy,
+  };
+
+  const className = cx({
+    [`${prefix}--switcher`]: true,
+    [customClassName]: !!customClassName,
+  });
+
+  const classNameItem = cx({
```
Comment (asudoh) [inline packages/react/src/components/UIShell/Switcher.js]:
```suggestion
  const classNameItem = cx(`${prefix}--switcher--item`, {
```

File: packages/react/src/components/UIShell/Switcher.js [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+    [`${prefix}--switcher--item`]: true,
+    [customClassName]: !!customClassName,
+  });
+
+  const classNameItemLink = cx({
+    [`${prefix}--switcher--item-link`]: true,
```
Comment (asudoh) [inline packages/react/src/components/UIShell/Switcher.js]:
```suggestion
```

File: packages/react/src/components/UIShell/Switcher.js [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    [`${prefix}--switcher`]: true,
+    [customClassName]: !!customClassName,
+  });
+
+  const classNameItem = cx({
+    [`${prefix}--switcher--item`]: true,
+    [customClassName]: !!customClassName,
+  });
+
+  const classNameItemLink = cx({
```
Comment (asudoh) [inline packages/react/src/components/UIShell/Switcher.js]:
```suggestion
  const classNameItemLink = cx(`${prefix}--switcher--item-link`, {
```

File: packages/react/src/components/UIShell/Switcher.js [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+    [`${prefix}--switcher--item-link`]: true,
+    [customClassName]: !!customClassName,
+  });
+
+  return (
+    <ul className={`${className}`} {...accessibilityLabel}>
```
Comment (asudoh) [inline packages/react/src/components/UIShell/Switcher.js]:
```suggestion
    <ul className={className} {...accessibilityLabel}>
```

File: packages/react/src/components/UIShell/Switcher.js [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+      {links &&
+        links.map((link, i) => {
+          return (
+            <li key={i} className={`${classNameItem}`}>
+              <a
+                className={`${classNameItemLink}`}
```
Comment (asudoh) [inline packages/react/src/components/UIShell/Switcher.js]:
```suggestion
                className={classNameItemLink}
```

**Review by asudoh — COMMENTED** [review]
Great to see the progress here @aledavila!

File: packages/react/src/components/UIShell/HeaderGlobalAction.js (hunk 1 above)
Comment (aledavila) [reply packages/react/src/components/UIShell/HeaderGlobalAction.js]: thanks no that was for testing I need to change it back.

File: packages/react/src/components/UIShell/HeaderPanel.js [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import { AriaLabelPropType } from '../../prop-types/AriaPropTypes';
+
+const { prefix } = settings;
+
+const HeaderPanel = React.forwardRef(function HeaderPanel(props, ref) {
+  const {
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledBy,
```
Comment (emyarod) [inline packages/react/src/components/UIShell/HeaderPanel.js:19]: are these aliases still needed if we can just use the aria attributes directly?

Comment (aledavila) [reply packages/react/src/components/UIShell/HeaderPanel.js:19]: I was going by the code already there, since the existing components already have them.

**Review by dakahn — APPROVED** [review]

**Review by jnm2377 — APPROVED** [review]

**Review by dakahn — APPROVED** [review]

**Review by asudoh — APPROVED** [review]
LGTM 👍 once visual review completes - Thanks @aledavila!

**Review by aagonzales — COMMENTED** [review]
Looking good!

We just need to update the interactive colors for the link in the "content switcher" like we did in this issue: https://github.com/carbon-design-system/carbon/issues/2805. Sorry, I probably should have had an issue for the right nav updates too.

Hover background: Gray 100 Hover / `#2c2c2c` 
Active background: Gray 80 / `#3d3d3d`
Selected background: Gray 90 / `#252525`

**Review by tw15egan — APPROVED** [review]
Looks good once design signs off!

**Review by dakahn — APPROVED** [review]

**Review by aagonzales — APPROVED** [review]
Perfect! Thanks for updating.

**Pull request merged**